### DISCUSSION TITLE
Howto PGP/MIME

### DISCUSSION THREAD

**Discussion by hagemic** [pgpainless/pgpainless#442] [Q&A]
Unfortunately, I can't find any sample code for creating an e-mail based on PGP/MIME. Can anyone help me with this?

**Comment by vanitasvitae** [comment]
Hey!
PGPainless is only an OpenPGP implementation. It does not handle PGP/MIME.
You can of course use PGPainless to create signed / encrypted body parts and then pack those into your MIME framework of choice.
Personally I have never done this though, so you'll need to figure it out on your own :)

**Comment by DenBond7** [comment]
> Unfortunately, I can't find any sample code for creating an e-mail based on PGP/MIME. Can anyone help me with this?

As @vanitasvitae said this question is not related to PGPainless directly. It will be better to ask on [StackOverflow](https://stackoverflow.com/)

But I can give you a little hint here))

To resolve your issue you can use [JavaMail](https://github.com/jakartaee/mail-api) to create a MIME message(read [FAQ](https://jakartaee.github.io/mail-api/FAQ.html#anytype)) and follow [MIME Security with OpenPGP](https://datatracker.ietf.org/doc/html/rfc3156#section-4) to fit open standards(the MIME message structure)
